tests/ttnn/unit_tests/operations/eltwise/test_binary_bcast.py

```diff
@@ -17,21 +21,90 @@
         (torch.Size([5, 1, 1, 64]), torch.Size([1, 3, 128, 1])),
     ),
 )
-def test_binary_scalar_ops(input_shapes, device):
+@pytest.mark.parametrize(
+    "ttnn_fn",
+    [
+        ttnn.experimental.gte,
+        ttnn.experimental.gt,
+        ttnn.experimental.lte,
+        ttnn.experimental.lt,
+        ttnn.experimental.eq,
+        ttnn.experimental.ne,
+        ttnn.experimental.logical_and,
+        ttnn.experimental.logical_or,
+        ttnn.experimental.logical_xor,
+        ttnn.experimental.ldexp,
+        ttnn.experimental.logaddexp,
+        ttnn.experimental.logaddexp2,
+        ttnn.experimental.squared_difference,
+        ttnn.experimental.add,
+        ttnn.experimental.sub,
+        ttnn.experimental.mul,
+        ttnn.experimental.div,
+        ttnn.experimental.bias_gelu,
+    ],
+)
+def test_binary_scalar_ops(input_shapes, ttnn_fn, device):
     a_shape, b_shape = input_shapes
     a_pt = torch.rand(a_shape).bfloat16()
     b_pt = torch.rand(b_shape).bfloat16()
 
     a_tt = ttnn.from_torch(a_pt, device=device, layout=ttnn.TILE_LAYOUT, memory_config=ttnn.DRAM_MEMORY_CONFIG)
     b_tt = ttnn.from_torch(b_pt, device=device, layout=ttnn.TILE_LAYOUT, memory_config=ttnn.DRAM_MEMORY_CONFIG)
     cq_id = 0
-    out_tt = ttnn.experimental.add(a_tt, b_tt, queue_id=cq_id)
-    out_pt = a_pt + b_pt
+    out_tt = ttnn_fn(a_tt, b_tt, queue_id=cq_id)
+    golden_fn = ttnn.get_golden_function(ttnn_fn)
+    out_pt = golden_fn(a_pt, b_pt)
 
     comp_pass = compare_pcc([out_tt], [out_pt])
     assert comp_pass
 
 
+@pytest.mark.parametrize(
+    "input_shapes",
+    (
+        (torch.Size([1, 1, 31, 32]), torch.Size([5, 3, 32, 32])),
+        (torch.Size([5, 2, 64, 1]), torch.Size([1, 3, 1, 128])),
+        (torch.Size([5, 1, 1, 64]), torch.Size([2, 3, 128, 1])),
```

Comment:
If possible, lets please also check with 2D, 3D, 5D shapes

Reply:
To my understanding, 4D tensors are currently expected as inputs for the purposes of broadcasting. Each dimension is independently handled for broadcasting logic as shown with the test-cases for `test_binary_scalar_ops`. We can view 2D, 3D, or 5D shapes as 4D without allocating extra memory, so that should suffice for 2D and 3D. If broadcasting along more than 4 dimensions is necessary, that can be implemented as an extension, but not as part of this PR.

Reply:
Please check out traces coming from PyTorch. 
https://github.com/tenstorrent/pytorch2.0_ttnn/blob/main/docs/operations/aten.add.Tensor.md

It is possible to receive tensors of different ranks, like rank_a 3 and rank_b 1. The expectation is that it works and users does not have to unsqueeze to 4D manually. 

If it does not work in this PR and requires more work, I agree that’s a matter for a different PR.

Reply:
This should work correctly without an unsqeeze.  If we find that this is not the case it should be fixed asap on the next PR.   The only edge case we don't support is the 5D tensors.   If generality requirements dictate we need this it should be prioritized.   My current thought is that the 5D support is lower priority than sharding in the context of broadcasting which is what I want us to focus on as soon as possible.